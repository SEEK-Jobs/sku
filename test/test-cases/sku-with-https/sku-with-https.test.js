const path = require('path');
const { promisify } = require('util');
const rimrafAsync = promisify(require('rimraf'));
const fs = require('fs-extra');
const runSkuScriptInDir = require('../../utils/runSkuScriptInDir');
const { getAppSnapshot } = require('../../utils/appSnapshot');
const waitForUrls = require('../../utils/waitForUrls');
const { getPathFromCwd } = require('../../../lib/cwd');

const { port, serverPort } = require('./app/sku-server.config');
const appDir = path.resolve(__dirname, 'app');

async function createPackageLink(name) {
  await fs.mkdirp(`${__dirname}/app/node_modules`);
  await fs.symlink(
    getPathFromCwd(`node_modules/${name}`),
    `${__dirname}/app/node_modules/${name}`,
  );
}

async function setUpLocalDependencies() {
  const nodeModules = `${__dirname}/app/node_modules`;
  await rimrafAsync(nodeModules);
  await Promise.all(['react', 'react-dom'].map(createPackageLink));
}

describe('sku-with-https', () => {
  beforeAll(async () => {
    // "Install" React and braid-design-system into this test app so that webpack-node-externals
    // treats them correctly.
    await setUpLocalDependencies();
  });

  describe('start', () => {
    const url = `https://localhost:${port}`;
    let process;

    beforeAll(async () => {
      process = await runSkuScriptInDir('start', appDir);
      await waitForUrls(url, `${url}/test-middleware`);
    });

    afterAll(async () => {
      await process.kill();
    });

    it('should start a development server', async () => {
      const snapshot = await getAppSnapshot(url);
      expect(snapshot).toMatchSnapshot();
    });
    it('should support the supplied middleware', async () => {
      const snapshot = await getAppSnapshot(`${url}/test-middleware`);
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('start-ssr', () => {
    const url = `https://localhost:${serverPort}`;
    let process;

    beforeAll(async () => {
      process = await runSkuScriptInDir('start-ssr', appDir, [
        '--config=sku-server.config.js',
      ]);
      await waitForUrls(url, `${url}/test-middleware`);
    });

    afterAll(async () => {
      await process.kill();
    });

    it('should support the supplied middleware', async () => {
      const snapshot = await getAppSnapshot(`${url}/test-middleware`);
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('serve', () => {
    const url = `https://localhost:${port}`;
    let process;

    beforeAll(async () => {
      await runSkuScriptInDir('build', appDir);
      process = await runSkuScriptInDir('serve', appDir);
      await waitForUrls(url, `${url}/test-middleware`);
    });

    afterAll(async () => {
      await process.kill();
    });

    it('should start a development server', async () => {
      const snapshot = await getAppSnapshot(url);
      expect(snapshot).toMatchSnapshot();
    });

    it('should support the supplied middleware', async () => {
      const snapshot = await getAppSnapshot(`${url}/test-middleware`);
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('.gitignore', () => {
    it('should add the .ssl directory to .gitignore', async () => {
      const ignoreContents = await fs.readFile(
        path.join(appDir, '.gitignore'),
        'utf-8',
      );
      expect(ignoreContents.split('\n')).toContain(`.ssl`);
    });
  });
});
