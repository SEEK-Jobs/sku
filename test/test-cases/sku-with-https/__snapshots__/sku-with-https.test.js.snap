// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sku-with-https serve should start a development server 1`] = `
SCRIPTS: Array [
  "/runtime-554e53c2f72903d6f45f.js",
  "/vendors~main-a264bd1d24dfb67ae5a1.js",
  "/main-147106e0f69abb56387e.js",
]
CSS: Array [
  "/main-fadb2269a374a06075a5.css",
]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta http-equiv="Content-Security-Policy"
          content="script-src 'self' https://some-cdn.com;"
    >
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="stylesheet"
          href="styles[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
    >
  </head>
  <body>
    <div id="app">
      <div class="_1brbpTo"
           data-reactroot
      >
        <div class="_3YtmnHj">
          Hello World
        </div>
        <img src="/a85d2645cc4d80a20b5ba78e8e797449.png">
      </div>
    </div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
    >
      []
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;

exports[`sku-with-https serve should support the supplied middleware 1`] = `
"OK"
POST HYDRATE DIFFS: 
-OK
\\ No newline at end of file
+<html>
+  <head>
+  </head>
+  <body>
+    OK
+  </body>
+</html>
\\ No newline at end of file

`;

exports[`sku-with-https start should start a development server 1`] = `
SCRIPTS: Array [
  "/runtime.js",
  "/vendors~devServerOnly~main.js",
  "/vendors~main.js",
  "/main.js",
]
CSS: Array [
  "/main.css",
]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta http-equiv="Content-Security-Policy"
          content="script-src 'self' https://some-cdn.com;"
    >
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="stylesheet"
          href="styles[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[3]"
    >
  </head>
  <body>
    <div id="app">
      <div class="lessStyles__root___1brbpTo"
           data-reactroot
      >
        <div class="lessStyles__nested___3YtmnHj">
          Hello World
        </div>
        <img src="/a85d2645cc4d80a20b5ba78e8e797449.png">
      </div>
    </div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
    >
      []
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[3]"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;

exports[`sku-with-https start should support the supplied middleware 1`] = `
"OK"
POST HYDRATE DIFFS: 
-OK
\\ No newline at end of file
+<html>
+  <head>
+  </head>
+  <body>
+    OK
+  </body>
+</html>
\\ No newline at end of file

`;

exports[`sku-with-https start-ssr should start a development server 1`] = `
SCRIPTS: Array [
  "/runtime-554e53c2f72903d6f45f.js",
  "/vendors~main-a264bd1d24dfb67ae5a1.js",
  "/main-147106e0f69abb56387e.js",
]
CSS: Array [
  "/main-fadb2269a374a06075a5.css",
]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta http-equiv="Content-Security-Policy"
          content="script-src 'self' https://some-cdn.com;"
    >
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="stylesheet"
          href="styles[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
    >
  </head>
  <body>
    <div id="app">
      <div class="_1brbpTo"
           data-reactroot
      >
        <div class="_3YtmnHj">
          Hello World
        </div>
        <img src="/a85d2645cc4d80a20b5ba78e8e797449.png">
      </div>
    </div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
    >
      []
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;

exports[`sku-with-https start-ssr should support the supplied middleware 1`] = `
SCRIPTS: Array [
  "https://localhost:9843/runtime.js",
  "https://localhost:9843/vendors~main.js",
  "https://localhost:9843/main.js",
]
CSS: Array [
  "https://localhost:9843/main.css",
]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <meta http-equiv="Content-Security-Policy"
          content="script-src 'self' https://localhost:9843 https://some-cdn.com;"
    >
    <link data-chunk="main"
          rel="stylesheet"
          href="styles[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
          crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <div class="lessStyles__root___1brbpTo">
        <div class="lessStyles__nested___3YtmnHj">
          Hello World
        </div>
        <img src="https://localhost:9843/a85d2645cc4d80a20b5ba78e8e797449.png">
      </div>
    </div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
            crossorigin="anonymous"
    >
      []
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
            crossorigin="anonymous"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;
